src/memory.js

```diff
@@ -1,18 +1,42 @@
 class MemoryGame {
   constructor(cards) {
     this.cards = cards;
-    // add the rest of the class properties here
+    this.pickedCards = []
+    this.pairsClicked = 0
+    this.pairsGuessed = 0
   }
 
   shuffleCards() {
-    // ... write your code here
+    if (this.cards === undefined) {
+      return undefined
+    }
+
+    for (let i = 0; i < this.cards.length; i++) {
+      const card = this.cards[i]
+      let random = Math.floor(Math.random() * this.cards.length)
+      this.cards[i] = this.cards[random]
+      this.cards[random] = card
+    }
   }
 
   checkIfPair(card1, card2) {
-    // ... write your code here
+    
+    this.pairsClicked++
+
+    if (card1 === card2) {
+      this.pairsGuessed++
+      return true
+    } else {
+      return false
+    }
```

Comment:
Esto esta bien pero yo haria esto otro
```suggestion
    if (card1 === card2) {
      this.pairsGuessed++
    }
    return card1 === card2
```

Es lo mismo :D, pero lo que has peusto tambien esta perfect!